app/controllers/energy_tariffs/energy_tariff_charges_controller.rb

```diff
@@ -22,11 +22,7 @@ def create
           @energy_tariff.energy_tariff_charges.destroy_all
           @energy_tariff_charges.each(&:save!)
         end
-        case @energy_tariff.tariff_holder_type
-        when 'School' then redirect_to school_energy_tariff_path(@school, @energy_tariff)
-        when 'SchoolGroup' then redirect_to school_group_energy_tariff_path(@school_group, @energy_tariff)
-        when 'SiteSettings' then redirect_to admin_settings_energy_tariff_path(@energy_tariff)
-        end
+        redirect_to polymorphic_path(@energy_tariff.tariff_holder_route + [@energy_tariff])
```

Comment:
This looks much neater! Looking across how you've used it, there seem to be the following varieties of call to polymorphic_path:

- just the tariff `... + [@energy_tariff]`
- with an extra path segment `... + [@energy_tariff, :energy_tariff_differential_prices]`
- with an id `... + [@energy_tariff, :energy_tariff_flat_price], id: @energy_tariff.energy_tariff_prices.first`
- with an action `action: 'edit'`
- with an action and an id

Wondering if you can wrap those all into an `energy_tariffs_path` helper, e.g:

```
def energy_tariffs_path(energy_tariff, path=[], options={})
  polymorphic_path(@energy_tariff.tariff_holder_route + [@energy_tariff] + path, options)
end
```

We do something similar for `advice_page_path`.

Then you could use:

```
energy_tariff_path(@energy_tariff, :energy_tariff_differential_prices)
energy_tariff_path(@energy_tariff, :energy_tariff_differential_price, id: 123)
energy_tariff_path(@energy_tariff, action: :choose_type)
```